### PR TITLE
Changes to java-spring-boot2 stack to facilitate downstream selection of snowdrop

### PR DESCRIPTION
**Pull request by richard-trotter**
### Checklist:

- [X] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [X] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [X] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stacks-overview.md#stack-structure).

### Modifying an existing stack:

- [X] Updated the stack version in `stack.yaml`

### Contributing a new stack:

- Describe how application dependencies are managed:

- Explain how Appsody file watcher is utilized:

- Describe other Appsody environment variables defined by the stack image:

- Describe any limitations and known issues:

### CODE REVIEW

File: incubator/java-spring-boot2/stack.yaml [hunk 1]
```diff
@@ -16,7 +16,10 @@ requirements:
    docker-version: ">= 17.09.0"
 templating-data:
   baseimage: maven:3.6-ibmjava-8
+  baseextrapkgs: curl wget xmlstarlet unzip ca-certificates
```
Comment (ebullient) [inline incubator/java-spring-boot2/stack.yaml]: I am having second thoughts about how much is in this string. If xmlstartlet is removed, the whole stack will break. We use it everywhere. I'm not sure this should be moved out this far...

File: incubator/java-spring-boot2/image/project/update.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -x
+
+XTRACMDS=$*
+
```
Comment (ebullient) [inline incubator/java-spring-boot2/image/project/update.sh]: Could still test for 'dev' vs. 'final' or something, but I think stack-required packages, like xmlstartlet, should be in here, not in stack.yaml. Too central to how mechanics of the stack work.

**Comment by richard-trotter** [comment]
@neeraj-laad Please note: ^^^

**Comment by richard-trotter** [comment]
@neeraj-laad I see a check remains outstanding: 

> license/cla Expected — Waiting for status to be reported

However it is noted above:

> All committers have signed the CLA.

Please advise on what I need to do next.

**Pull request merged**